### DISCUSSION TITLE
Perceptual gradient instead of linear

### DISCUSSION THREAD

**Discussion by matze** [hecrj/iced#2043] [Ideas]
I noticed that iced computes the gradient in the ~linear~ sRGB color space which makes for some very unpleasant gradients when going from very bright to very dark colors. This issue has been discussed plenty of times (see [Björn Ottosson's post](https://bottosson.github.io/posts/colorwrong) for an explanation and [Raph Levien's interactive tool](https://raphlinus.github.io/color/2021/01/18/oklab-critique.html) for alternative color space comparisons) and it seems that [Oklab](https://bottosson.github.io/posts/oklab/) has become somewhat of a defacto standard. So my question is, if the current implementation could be changed or at least a second gradient implementation be added?

**Comment by matze** [comment]
I have made a little PoC change to `quad.wgsl` basically porting this [shadertoy example](https://www.shadertoy.com/view/ttcyRS). The difference looks as follows (top is Oklab, bottom the current gradient computation):
 
![oklab-vs-srgb](https://github.com/iced-rs/iced/assets/115270/845b8713-5ea7-4ac5-8f14-d5e57a439d58)

The white to black transition quite clearly shows the different perceptually smoother transition. So again, is that of interest?

**Reply by bungoboingo** [reply]
Hey! I'm definitely interested in adding OkLab support to our existing gradient implementation, or just using it as a default and removing linear blending entirely. We do have the `web-colors` flag now to force SRGB blending for web parity, OkLab color space blending would have to exist somewhere alongside that. Since `web-colors` is a compilation flag we could use a separate fragment shader for each implementation if needed. Very interested in seeing a PR!

**Comment by hecrj** [comment]
Important to note here that `iced` computes gradients in linear RGB, unless the `web-colors` feature is enabled.

This is subjective, but in that example I personally prefer the bottom version. Are you sure top is OkLab?

**Reply by bungoboingo** [reply]
Ah, I misread the original comment, thought bottom was OkLab. I also prefer the bottom gradient!

**Comment by matze** [comment]
> Are you sure top is OkLab?

It is but I wrongly de-gamma'd before and after leading to artifacts. Here it is without that, again top OkLab, bottom as it is now with iced 0.10 (sorry for flipped text, wanted to have the same arrangement as above):

![oklab-vs-srgb-2](https://github.com/iced-rs/iced/assets/115270/880e8a03-5deb-4f92-a10e-fd3391379b01)

Yes, it's less pronounced for certain color combinations but still very visible for black to white (smoother transition) and yellow to blue (no hue shift). You can check with Raph Levien's tool (make sure to expand the list of color spaces to get access to linear) how gradients compute, it matches pretty much 1:1.
